### PR TITLE
Cleanup MSBuild conversion and unify dependency versions

### PR DESCRIPTION
**Pull request by natemcmaster**
cc @mikaelm12

### CODE REVIEW

File: src/Microsoft.AspNetCore.Proxy/Microsoft.AspNetCore.Proxy.csproj [hunk 1]
```diff
@@ -11,10 +11,10 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.WebSockets" Version="1.1.0-*" />
-    <PackageReference Include="Microsoft.Extensions.Options" Version="1.2.0-*" />
-    <PackageReference Include="System.Net.WebSockets.Client" Version="4.4.0-*" />
-    <PackageReference Include="System.Net.Http" Version="4.4.0-*" />
+    <PackageReference Include="Microsoft.AspNetCore.WebSockets" Version="$(WebSocketsVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Options" Version="$(AspNetCoreVersion)" />
+    <PackageReference Include="System.Net.WebSockets.Client" Version="$(CoreFxVersion)" />
+    <PackageReference Include="System.Net.Http" Version="$(CoreFxVersion)" />
```
Comment (mikaelm12) [inline src/Microsoft.AspNetCore.Proxy/Microsoft.AspNetCore.Proxy.csproj]: System.Net.Http is brought in by NetStandard 1.6.1

**Comment by natemcmaster** [comment]
cc @lajones this should resolve the CI failures

File: src/Microsoft.AspNetCore.Proxy/Microsoft.AspNetCore.Proxy.csproj (hunk 1 above)
Comment (mikaelm12) [inline src/Microsoft.AspNetCore.Proxy/Microsoft.AspNetCore.Proxy.csproj]: System.Net.Http is brought in by NetStandard 1.6.1 so we don't need this line

Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Proxy/Microsoft.AspNetCore.Proxy.csproj]: ok, will remove.

**Comment by mkosieradzki** [comment]
Please care for System.Net.Http there is a substantial difference between 4.3.1 and 4.3!

**Comment by natemcmaster** [comment]
What's the difference between 4.3.0 and 4.3.1?

**Comment by mkosieradzki** [comment]
@natemcmaster 

> What's the difference between 4.3.0 and 4.3.1?

AFAIR .NET Standard 1.6.1 has 4.3.0 and we use 4.3.1

4.3.0 is total disaster in terms of binding redirect requirements

4.3.1 is much better, but! It has been reverted to the ground (4.0.0) in .NET 4.6.

So in #49 I needed to conditionally use WinHttpHandler to avoid #47 .

**Comment by mkosieradzki** [comment]
And also I would really appreciate merging #46 before upgrading versions, as I have stabilized it (and cleanup dependencies as well) it is ready to merge.

**Comment by natemcmaster** [comment]
@mkosieradzki we're going to push forward with this before #46 as our CI is currently failing. The last PR we merged broke things.

**Comment by mkosieradzki** [comment]
@natemcmaster Fair enough! I will merge later then! BTW. I am waiting for the fixed CI in #49 . :)

**Pull request merged**